src/textual/widgets/_option_list.py

```diff
@@ -0,0 +1,943 @@
+
+        Returns:
+            The `OptionList` instance.
+
+        Raises:
+            OptionDoesNotExist: If there is no option with the given index.
+        """
+        try:
+            return self._set_option_disabled(index, False)
+        except IndexError:
+            raise OptionDoesNotExist(
+                f"There is no option with an index of {index}"
+            ) from None
+
+    def disable_option_at_index(self, index: int) -> Self:
+        """Disable the option at the given index.
+
+        Returns:
+            The `OptionList` instance.
+
+        Raises:
+            OptionDoesNotExist: If there is no option with the given index.
+        """
+        try:
+            return self._set_option_disabled(index, True)
+        except IndexError:
+            raise OptionDoesNotExist(
+                f"There is no option with an index of {index}"
+            ) from None
+
+    def enable_option(self, option_id: str) -> Self:
+        """Enable the option with the given ID.
+
+        Args:
+            option_id: The ID of the option to enable.
+
+        Returns:
+            The `OptionList` instance.
+
+        Raises:
+            OptionDoesNotExist: If no option has the given ID.
+        """
+        try:
+            return self.enable_option_at_index(self._option_ids[option_id])
+        except KeyError:
+            raise OptionDoesNotExist(
+                f"There is no option with an ID of '{option_id}'"
+            ) from None
+
+    def disable_option(self, option_id: str) -> Self:
+        """Disable the option with the given ID.
+
+        Args:
+            option_id: The ID of the option to disable.
+
+        Returns:
+            The `OptionList` instance.
+
+        Raises:
+            OptionDoesNotExist: If no option has the given ID.
+        """
+        try:
+            return self.disable_option_at_index(self._option_ids[option_id])
+        except KeyError:
+            raise OptionDoesNotExist(
+                f"There is no option with an ID of '{option_id}'"
+            ) from None
+
+    @property
+    def option_count(self) -> int:
+        """The count of options."""
+        return len(self._options)
+
+    @property
+    def options(self) -> Iterator[Option]:
+        """An iterator of options within the list."""
+        return iter(self._options)
+
+    def get_option_at_index(self, index: int) -> Option:
+        """Get the option at the given index.
+
+        Args:
+            index: The index of the option to get.
+
+        Returns:
+            The option at that index.
+
+        Raises:
+            OptionDoesNotExist: If there is no option with the index.
+        """
+        try:
+            return self._options[index]
+        except IndexError:
+            raise OptionDoesNotExist(
+                f"There is no option with an index of {index}"
+            ) from None
+
+    def get_option(self, option_id: str) -> Option:
```

Comment:
I see there are a lot of `do_something_option(_at_index)`.
Would it be messy if all those pairs were collapsed into a single method that just received an index or an ID, and then retrieved the appropriate option?
@davep @willmcgugan